R/validate_inputs.R

```diff
@@ -0,0 +1,164 @@
+
+#' send input validation messages to output
+#'
+#' Captures messages from `InputValidator` objects and collates them
+#' into one message passed to `validate`.
+#'
+#' `shiny::validate` is used to withhold rendering of an output element until
+#' certain conditions are met and a print a validation message in place
+#' of the output element.
+#' `shinyvalidate::InputValidator` allows to validate input elements
+#' and display specific messages in their respective input widgets.
+#' This function is a hybrid solution. Given an `InputValidator` object,
+#' it extracts messages from inputs that fail validation and places them all in one
+#' validation message that is passed to a `validate`/`need` call.
+#' This way the input validator messages are repeated in the output.
+#'
+#' \code{validate_inputs} accepts an arbitrary number of `InputValidator`s
+#' and prints all messages together, adding one (optional) header.
+#' \code{validate_inputs_segregated} accepts a list of `InputValidator`s
+#' and prints messages grouped by validator. If elements of \code{validators} are named,
+#' the names are used as headers for their respective message groups.
+#'
+#'
+#' @name validate_inputs
+#'
+#' @param ... for \code{validate_inputs} any number of `InputValidator` objects \cr
+#'            for \code{validate_inputs_segregated} arguments passed to `shiny::validate`
+#' @param header `character(1)` optional generic validation message
+#' @param validators optionally named `list` of `InputValidator` objects, see \code{Details}
+#'
+#' @return
+#' Returns NULL if the final validation call passes and a `shiny.silent.error` if it fails.
+#'
+#' @seealso \code{[shinyvalidate::InputValidator]} \code{[shiny::validate]}
```

Comment:
```suggestion
#' @seealso [`shinyvalidate::InputValidator`],  [`shiny::validate`]
```

I think we can do this?

Reply:
Yes! Thank you! I could NOT get the links to work 😠